statsmodels/discrete/count_model.py

```diff
@@ -0,0 +1,468 @@
+            offset = getattr(self, "offset", 0) + getattr(self, "exposure", 0)
+            if np.size(offset) == 1 and offset == 0:
+                offset = None
+            mod_poi = Poisson(self.endog, self.exog, offset=offset)
+            start_params = mod_poi.fit_regularized(
+                start_params=start_params, method=method, maxiter=maxiter,
+                full_output=full_output, disp=0, callback=callback,
+                alpha=alpha_p, trim_mode=trim_mode, auto_trim_tol=auto_trim_tol,
+                size_trim_tol=size_trim_tol, qc_tol=qc_tol, **kwargs).params
+            start_params = np.append(start_params, 0.1)
+        cntfit = super(CountModel, self).fit_regularized(
+                start_params=start_params, method=method, maxiter=maxiter,
+                full_output=full_output, disp=disp, callback=callback,
+                alpha=alpha, trim_mode=trim_mode, auto_trim_tol=auto_trim_tol,
+                size_trim_tol=size_trim_tol, qc_tol=qc_tol, **kwargs)
+
+        if method in ['l1', 'l1_cvxopt_cp']:
+            discretefit = self.result_reg(self, cntfit)
+        else:
+            raise Exception(
+                    "argument method == %s, which is not handled" % method)
+
+        return self.result_reg_wrapper(discretefit)
+
+    fit_regularized.__doc__ = DiscreteModel.fit_regularized.__doc__
+
+    def score(self, params):
+        """
+        Generic Zero Inflated model score (gradient) vector of the log-likelihood
+
+        Parameters
+        ----------
+        params : array-like
+            The parameters of the model
+
+        Returns
+        -------
+        score : ndarray, 1-D
+            The score vector of the model, i.e. the first derivative of the
+            loglikelihood function, evaluated at `params`
+        """
+        params_infl = params[:self.k_inflate]
+        params_main = params[self.k_inflate:]
+
+        y = self.endog
+        w = self.model_infl.predict(params_infl)
+        w[w == 1.] = np.nextafter(1, 0)
+        score_main = self.model_main.score_obs(params_main)
+        llf_main = self.model_main.loglikeobs(params_main)
+        llf = self.loglikeobs(params)
+        zero_idx = np.nonzero(y == 0)[0]
+        nonzero_idx = np.nonzero(y)[0]
+
+        mu = self.model_main.predict(params_main)
+
+        coeff = (1 + w[zero_idx] * (np.exp(mu[zero_idx]) - 1))
+        dldp_zero = (score_main[zero_idx].T *
+                     (1 - (w[zero_idx]) / np.exp(llf[zero_idx]))).T.sum(0)
+        dldp_nonzero = score_main[nonzero_idx].sum(0)
+        dldp = dldp_zero + dldp_nonzero
+
+        dldw_zero =  self.exog_infl[zero_idx].T.dot(w[zero_idx] *
+            (1 - w[zero_idx]) * (1 - np.exp(llf_main[zero_idx])) / np.exp(llf[zero_idx]))
+        dldw_nonzero = -self.exog_infl[nonzero_idx].T.dot(w[nonzero_idx])
+        dldw = dldw_zero + dldw_nonzero
+
+        return np.concatenate((dldw, dldp))
+
+    def _hessian_partly(self, params):
```

Comment:
Another thought, you could use numerical derivatives just for this part, if the rest can be implemented analytically.